pkg/controllers/allocation/controller.go

```diff
@@ -68,7 +68,7 @@ func NewController(kubeClient client.Client, coreV1Client corev1.CoreV1Interface
 		Filter:        &Filter{KubeClient: kubeClient},
 		Binder:        DecorateBinderMetrics(NewBinder(kubeClient, coreV1Client)),
 		Batcher:       NewBatcher(maxBatchWindow, batchIdleTimeout),
-		Scheduler:     scheduling.NewScheduler(cloudProvider, kubeClient),
+		Scheduler:     scheduling.DecorateSchedulerMetrics(scheduling.NewScheduler(cloudProvider, kubeClient)),
```

Comment:
thoughts on encapsulating this inside NewScheduler so the test doesn't need to be aware it's decorated?

Reply:
That's a good point, I agree that this may not be the best place to apply the metrics decoration.  However, I would prefer pushing the metrics decoration up to the caller of `allocation.NewController()` rather than down into `scheduling.NewScheduler()`.

Reply:
It depends whether or not you think the scheduler should innately emit metrics (i.e., the decorator is a convenience impl detail) or whether something should be decorating it. I'd lean the former, as I don't see multiple scheduler implementations in our future. This does feel somewhat like go/java philosophies colliding.

Nonblocking, though.